mods/doors/init.lua

```diff
@@ -1,435 +1,440 @@
-doors = {}
 
--- Registers a door
-function doors.register_door(name, def)
-	def.groups.not_in_creative_inventory = 1
+--[[
 
-	local box = {{-0.5, -0.5, -0.5, 0.5, 0.5, -0.5+1.5/16}}
+Copyright (C) 2012 PilzAdam
+  modified by BlockMen (added sounds, glassdoors[glass, obsidian glass], trapdoor)
+Copyright (C) 2015 - Auke Kok <sofar@foo-projects.org>
 
-	if not def.node_box_bottom then
-		def.node_box_bottom = box
-	end
-	if not def.node_box_top then
-		def.node_box_top = box
-	end
-	if not def.selection_box_bottom then
-		def.selection_box_bottom= box
+--]]
+
+-- our API object
+doors = {}
+
+-- private data
+_doors = {}
+_doors.registered_doors = {}
+_doors.registered_trapdoors = {}
+
+-- returns an object to a door object or nil
+doors.get = function(pos)
```

Comment:
Minor nitpicking, but can the `foo.bar = function(...)` form just die already? `function foo.bar(...)` does exactly the same and looks prettier <3